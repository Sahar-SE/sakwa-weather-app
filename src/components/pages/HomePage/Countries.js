import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Country from './Country';
import { getCountries } from '../../../redux/CountriesState/countriesState';

const Countries = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getCountries());
  }, []);
  const countries = useSelector((state) => (state.countries));
  return (
    <div className="countriesPage" data-testid="countriesContent">
      <div className="homePageCover">
        <div className="title">
          <h1>Weather Condition</h1>
          <p>Select the country you wanna see its weather</p>
        </div>
      </div>
      <div className="padding">
        <h2 className="countriesHeader">Countries List</h2>
      </div>
      <div className="padding countries">
        {Object.values(countries).map((country) => (
          <Country
            key={country.iso3}
            name={country.name}
            states={country.states}
            iso3={country.iso3}
          />
        ))}
      </div>
      <small>developed by SaharSaba Amiri</small>
    </div>
  );
};

export default Countries;
