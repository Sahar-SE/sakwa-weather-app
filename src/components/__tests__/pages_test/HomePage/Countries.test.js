import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../../redux/configureStore';

import Countries from '../../../pages/HomePage/Countries';

describe('Testing Countries component', () => {
  render(
    <Provider store={store}>
      <Countries />
    </Provider>,
  );
  test('Testing the dom for success rendering elements on Countries component', () => {
    expect(screen.queryByTestId('countriesContent')).toBeInTheDocument();
  });

  test('Testing the dom for non existing elemente on the page after rendering', () => {
    expect(screen.queryByTestId('nothing')).not.toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Countries page="just for test">Countries</Countries>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
