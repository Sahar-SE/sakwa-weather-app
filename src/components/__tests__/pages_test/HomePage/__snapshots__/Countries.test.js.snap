// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Countries component renders correctly 1`] = `
<div
  className="countriesPage"
  data-testid="countriesContent"
>
  <div
    className="homePageCover"
  >
    <h1>
      World Weather
    </h1>
    <p>
      Choose the country
    </p>
  </div>
  <div
    className="padding"
  >
    <h2
      className="countriesHeader"
    >
      Countries
    </h2>
  </div>
  <div
    className="padding countries"
  />
</div>
`;
